pkg/config/env.go

```diff
@@ -212,4 +212,22 @@ var Config = struct {
 	// UI path  => localhost:18000/foo"
 	// API path => localhost:18000/foo/api/v1"
 	WebPrefix string `env:"FLAGR_WEB_PREFIX" envDefault:""`
+
+	// rules for validating flags on enable - when a flag is updated or enabled
+	// (putFlag or setFlagEnabled), the flag is evaluated against these
+	// optional rules. The operator indicates if all the rules must pass ('AND' value)
+	// or if at least one rule must pass ('OR' value)
+	//
+	// Example rule:
```

Comment:
If we introduce generic validation rules, I worried that we may need to carry on these validation rules' compatibility forever, and it's error-prone given that there're four layers of syntax for the Rules:

1. regex
2. flag's properties as the goevaluate context
3. custom goevaluate syntax like `any` / `all`
4. EnabledFlagValidationOperation for multiple rules

There's a steep learning curve without a place to try-and-test these rules - especially in production.

How about we start from the dedicated validation first? For example, we can have, for example

- `CustomTagsValidationRules`
- `CustomFlagKeyValidationRules`
- ...

Reply:
> There's a steep learning curve without a place to try-and-test these rules
To me, using the test framework seemed like a relatively good way to try out things: https://github.com/checkr/flagr/pull/431/files#diff-a5d5bc2d830216d981ead1c1e20ca3ab19c22874b17bb775fc3188d2992ded58R96

> How about we start from the dedicated validation first?
I think starting with `CustomTagsValidationRules` would handle the use case I have.  I'm a little unsure how it would handle if I had two different formats of tags that I wanted to enforce, and that seems like a common enough use case after you have one tag format.  Are the semantics of `CustomTagsValidationRules` that any tag would have to meet at least one of the validation rules? Because that's not what I want; if the flag has no tags it would pass the rules.  Are the semantics that a flag must have one tag that meets the rules? Then I can't enforce two different types of tags.  I guess I'd lean towards removing `4. EnabledFlagValidationOperation for multiple rules` and making `EnabledFlagValidationRules` a single string. 

> custom goevaluate syntax like any / all
The custom govaluate functions seem like something that should be in the govaluate library rather than us adding them here, but I could understand if they don't want to add in everything to the shared lib.  Still I should probably offer a PR.